File: Data Structures & Algorithms/concatenation-of-array/submission-0.py
class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ans = list()
        ans = [0] * (2 * n)
        for i, num in enumerate(nums):
            ans[i] = num
            ans[i + n] = num
        return ans

File: Data Structures & Algorithms/first-missing-positive/submission-0.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        for (int i = 0; i < nums.size(); i++) {
            while (nums[i] > 0 && nums[i] <= nums.size() && nums[i] != nums[nums[i] - 1]) {
                swap(nums[i], nums[nums[i] - 1]);
            }
        }
        int res = nums.size() + 1;
        for (int i = 0; i < nums.size(); i++) {
            if (i + 1 != nums[i]) {
                res = i + 1;
                break;
            }
        }
        return res;
    }
};